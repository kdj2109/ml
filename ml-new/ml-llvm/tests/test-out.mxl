void out(int[][][] m, int r, int c, int l) {
  int i;
  int j;
  int k;

  prints("P3");
  prints("255");
  printsl(r);
  printssl(" ");
  print(c);

  for (i = 0; i < r; i = i + 1) {
    for (j = 0; j < c; j = j + 1) {
      for (k = 0; k < l; k = k + 1) {
        printsl($m);
        printssl(" ");
        m = m.+;
      }
    }
    prints("");
  }
}

int main() {
  int[3][2:2] x;

  x = [| (255, 0, 0), (0, 255, 255)
       | (0, 255, 0), (255, 0, 255) |];

  out(@@@x, x.rows, x.columns, x.length);

  return 0;
}
