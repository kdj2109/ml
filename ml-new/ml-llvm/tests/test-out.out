P3
255
2 2
255 0 0 0 255 255
0 255 0 255 0 255
